### PR TITLE
Airport Challenge

### PR DESCRIPTION
**Pull request by tcpickard94**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,79 +1,13 @@
...
-* Write an RSpec **feature** test that lands and takes off a number of planes
-
-Note that is a practice 'tech test' of the kinds that employers use to screen developer applicants.  More detailed submission requirements/guidelines are in [CONTRIBUTING.md](CONTRIBUTING.md)
-
-Finally, don’t overcomplicate things. This task isn’t as hard as it may seem at first.
-
-* **Submit a pull request early.**  There are various checks that happen automatically when you send a pull request.  **Fix these issues if you can**.  Green is good.
-
-* Finally, please submit a pull request before Monday at 9am with your solution or partial solution.  However much or little amount of code you wrote please please please submit a pull request before Monday at 9am.
+https://travis-ci.org/tcpickard94/airport_challenge.svg?branch=master
```
Comment (katie210) [inline README.md]: Worth reviewing against this style/substance guide: http://stackoverflow.com/questions/2304863/how-to-write-a-good-readme

File: spec/airport_spec.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
+require 'airport'
+
+describe Airport do
+  let(:plane) { double(:plane) }
+  
+  it {expect(subject).to be_a Airport}
```
Comment (katie210) [inline spec/airport_spec.rb]: https://github.com/makersacademy/airport_challenge/blob/master/docs/review.md#use-named-subject-with-described_class

File: spec/airport_spec.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    
+    it {expect(subject).to respond_to(:landed).with(1).argument}
+    
+    it 'returns string if plane has landed' do
+      allow(subject).to receive(:weather) {:sunny}
+      expect(subject.landed(plane)).to eq 'Plane has touched down!!'
+    end
+    
+    it 'adds a plane to the airport' do
+      allow(subject).to receive(:weather) {:sunny}
```
Comment (katie210) [inline spec/airport_spec.rb]: Perhaps a before block could allow DRYer weather code: 
https://github.com/makersacademy/airport_challenge/blob/master/docs/review.md#use-before-blocks-to-set-up-objects-rather-than-repeat-code

File: spec/airport_spec.rb [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    end
+  end
+  
+  describe '#leave_gate' do
+    it {expect(subject).to respond_to(:leave_gate)}
+    
+    it 'Removes a plane from the airport' do
+      allow(subject).to receive(:weather) {:sunny}
+      subject.landed(plane)
+      expect(subject.leave_gate).to eq "Plane has taken off!!"
```
Comment (katie210) [inline spec/airport_spec.rb]: prefer allow/avoid double "expect" in it block

File: spec/airport_spec.rb [hunk 5]
```diff
@@ -0,0 +1,63 @@
+require 'airport'
+
+describe Airport do
+  let(:plane) { double(:plane) }
+  
+  it {expect(subject).to be_a Airport}  
+      
+  describe '#landed' do
+    
+    it {expect(subject).to respond_to(:landed).with(1).argument}
```
Comment (katie210) [inline spec/airport_spec.rb]: No need for this respond_to expectation when method is tested separately.

File: lib/airport.rb [hunk 6]
```diff
@@ -0,0 +1,33 @@
+class Airport
+  DEFAULT_CAPACITY = 15
+  attr_reader :planes
+  
+  def initialize(capacity = DEFAULT_CAPACITY)
+    @planes = []
```
Comment (katie210) [inline lib/airport.rb]: Perhaps review public methods for privatization:

https://github.com/makersacademy/airport_challenge/blob/master/docs/review.md#do-not-expose-internal-implementation

File: lib/airport.rb [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+  
+  def leave_gate
+    fail "It is too stormy to take off." if weather == :stormy
+    planes.pop
+    "Plane has taken off!!"
+  end
+  
+  private
+  
+  def weather
```
Comment (katie210) [inline lib/airport.rb]: SRP -- Maybe implement weather as its own class?

File: lib/plane.rb [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+class Plane
+
+  def initialize
+    @flying = false
+  end
+  
+  def land(airport) 
+    fail "Already on the ground" unless flying
+    @flying = false
+    airport.landed(self)
```
Comment (katie210) [inline lib/plane.rb]: Explore edge case of stormy weather preventing landing -- what happens to your plane? (@flying = true/false?)

File: lib/plane.rb [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+  end
+  
+  def take_off(airport)
+    fail "Already in the air" if flying
+    @flying = true
```
Comment (katie210) [inline lib/plane.rb]: https://github.com/makersacademy/airport_challenge/blob/master/docs/review.md#avoid-defining-attr_reader-then-accessing-the-instance-variable-directly

File: lib/airport.rb [hunk 10]
```diff
@@ -0,0 +1,33 @@
+class Airport
+  DEFAULT_CAPACITY = 15
+  attr_reader :planes
```
Comment (katie210) [inline lib/airport.rb]: Great example-- allow attr_reader :planes to guide other implementations of reader

**Comment by mechacoach** [comment]
You had your airport_challenge reviewed by **Kait Beaudette**

### The good points are:

* Overview of project in the README
* The solution avoids vacuous tests
* There are a minimal number of untested paths in the code
* Random behaviour is stubbed where appropriate
* The code follows Ruby naming conventions, Readability is supported by consistent naming
* Guard clauses are used where appropriate, Booleans are returned implicitly where appropriate, removeallcommented-outcode: There is no commented-out code in the submission
* Code avoids unnecessary public methods
* Each methods has a single responsibility
* Code avoids strings where it should use symbols
* The solution does not have any redundant code

### You should consider the following improvements:

* Installation Instructions should be in the README
* Use named subject with `described_class`
* Use `context` and `describe` blocks to create test scopes
* Use `before` blocks to set up objects rather than repeat code
* Avoid multiple `expect`s in `it` block
* Get rid of redundant `respond_to` expectations
* Remove all commented-out code
* Avoid magic numbers (e.g. on capacity)
* Follow Command/Query naming and separation
* Improper use of attr_*

### Additional comments:
It might be useful to consider the overarching logic of each method 
(i.e the airport doesn't control, but reads the weather, and while it doesn't fly/land it dictates that ability)., avoidmagicnumberse.g.oncapacity: Code avoids magic numbers

see https://github.com/makersacademy/airport_challenge/blob/master/docs/review.md for more details

**Comment by mechacoach** [comment]
You had your airport_challenge reviewed by **Kait Beaudette**

### The good points are:

* Overview of project in the README
* The solution avoids vacuous tests
* There are a minimal number of untested paths in the code
* Random behaviour is stubbed where appropriate
* The code follows Ruby naming conventions, Readability is supported by consistent naming
* Guard clauses are used where appropriate, Booleans are returned implicitly where appropriate, removeallcommented-outcode: There is no commented-out code in the submission
* Code avoids unnecessary public methods
* Each methods has a single responsibility
* Code avoids strings where it should use symbols
* The solution does not have any redundant code

### You should consider the following improvements:

* Installation Instructions should be in the README
* Use named subject with `described_class`
* Use `context` and `describe` blocks to create test scopes
* Use `before` blocks to set up objects rather than repeat code
* Avoid multiple `expect`s in `it` block
* Get rid of redundant `respond_to` expectations
* Remove all commented-out code
* Avoid magic numbers (e.g. on capacity)
* Follow Command/Query naming and separation
* Improper use of attr_*

### Additional comments:
It might be useful to consider the overarching logic of each method 
(i.e the airport doesn't control, but reads the weather, and while it doesn't fly/land it dictates that ability)., avoidmagicnumberse.g.oncapacity: Code avoids magic numbers

see https://github.com/makersacademy/airport_challenge/blob/master/docs/review.md for more details

**Pull request closed without merging**